pyrene/src/components/DateTimePicker/DateTimeInput/DateTimeInput.tsx

```diff
@@ -0,0 +1,139 @@
+import React, { forwardRef } from 'react';
+import clsx from 'clsx';
+
+import { allowedValueCheck } from '../../../utils/DateUtils';
+
+import Icon from '../../Icon/Icon';
+import styles from './dateTimeInput.css';
+
+
+export interface DateTimeInputProps {
+  /**
+   * Date format used by component
+   */
+  dateFormat?: string,
+  /**
+   * Boolean to toggle time display
+  */
+  dateOnly?: boolean,
+  /**
+   * Input date value
+  */
+  dateValue?: string,
+  /**
+   * Input error
+  */
+  errorValue?: string,
+  /**
+   * Component is disabled
+  */
+  disabled?: boolean,
+  /**
+   * Input field label
+  */
+  label?: string,
+  /**
+   * Input field name
+  */
+  name?: string,
+  /**
+   * Onchange from parent. Currently this handles onchange function passed from react-datepicker
+  */
+  onChange?: (event: React.ChangeEvent<HTMLInputElement>) => void,
+  /**
+   * Onclick from parent. Currently this handles onclick function passed from react-datepicker
+  */
+  onClick?: () => void,
+  /**
+   * Onfocus from parent.
+  */
+  onFocus?: () => void,
+  /**
+   * Callback to set Date value in parent
+   */
+  setDateValue?: (value: string) => void,
+  /**
+   * Callback to set Time value in parent
+   */
+  setTimeValue?: (value: string) => void,
+  /**
+   * Time format used by component
+   */
+  timeFormat?: string,
+  /**
+   * Input time value
+  */
+  timeValue?: string,
+}
+
+const DateTimeInput = forwardRef(({
+  dateFormat = 'dd.MM.yyyy',
+  dateOnly = false,
+  dateValue = '',
+  errorValue,
+  disabled,
+  label = dateOnly ? 'Date' : 'Date & Time',
+  name,
+  onChange,
+  onClick,
+  onFocus,
+  setDateValue,
+  setTimeValue,
+  timeFormat = ' HH:mm',
+  timeValue = '',
+}:DateTimeInputProps, ref:React.Ref<HTMLInputElement>) => {
+
+  const handleDateOnChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const node = event.target as HTMLInputElement;
+
+    if (allowedValueCheck(node.value)) {
+      if (!dateOnly && node.value.length > dateFormat.length) {
+        setDateValue?.(node.value.substring(0, dateFormat.length));
+        setTimeValue?.(node.value.substring(dateFormat.length));
```

Comment:
This I understand. But we should think of Mission Control first before other use cases.
Example
```
import parse from 'date-fns/parse';
const lg = parse('19.08.20', 'dd.MM.yyyy', new Date());
console.log(lg);
```
The year will be year 0020, not year 2000 or year 2020 and this could have very huge impacts when creating Service Windows for customers.
Users being able to specify their own formats was a later implementation. There are no mocks to support that. And maybe we could brainstorm on how to get this working properly.

Reply:
It has been decided that we stick with using the format length check and restricting the applicable formats